### DISCUSSION TITLE
Is the upgrade to Snakeyaml 2.0 safe?

### DISCUSSION THREAD

**Discussion by dave2wave** [apache/pulsar#20295] [General]
I noticed that Apache jclouds just reverted their upgrade to snakeyaml 2.0 - See
See https://github.com/apache/jclouds/commit/cf4a926afef2887d818c41d65a91a4a9c95a1c2

There are several PRs recently applied to Pulsar to do a similar upgrade. I think that these updates may be cherry picked which might be a problem.

**Comment by tisonkun** [comment]
I remember there is a jackson imcompatible issue. cc @Technoboy- @mattisonchao

**Comment by hezhangjian** [comment]
IIRC, jackson also upgrade to snakeyaml 2.0 in 2.15.0

**Reply by tisonkun** [reply]
Yes. You remind me of the whole story:

* Jackson 2.14.x doesn't bump to snakeyaml 2.0. Pulsar depends on 2.14.2 now. Then in some modules the snakeyaml version will be resolved to < 2.0.
* We ever bumped the version of jackson but later revert it due to regression:
  * https://github.com/apache/pulsar/pull/20177
  * https://github.com/apache/pulsar/pull/20182
